### PR TITLE
Expose ed25519 key explicitly and remove admin_key in t_entities

### LINKED ISSUE #56: Expose entity keys in the t_entities_table
Entity keys (key for accounts and files and admin_key for contracts) are currently stored in protobuf format.

1. Add column to store key in plain text
1.1 Consolidate key and admin_key into a single column
2. Modify java code to extract the ED25519 key from protobuf at the time of parsing and store in that column
3. Ignore keys that contain keylist or threshold keys
4. Optionally implement a migration for the entities that already exist in the database.

### PR DESCRIPTION
**Pull request by mike-burrage-hedera**
**Detailed description**:

key column will now be used in place of admin_key where admin_key was used in the past. No entity had both

- copied data from t_entities.admin_key to t_entities.key if present
- deprecated t_entities.admin_key
- added t_entities.ed25519_public_key_hex (lowercase hex string if the key)
- updated java, sql function, and rest-api for this change
- rest-api will no longer be returning admin_key publicly (didn't make sense on accounts anyway)

**Which issue(s) this PR fixes**:
Fixes #56 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [X] Tests updated

### CODE REVIEW

File: src/main/java/com/hedera/utilities/Utility.java [hunk 1]
```diff
@@ -745,4 +749,20 @@ public static File getResource(String path) {
...
+	 */
+	public @Nullable String protobufKeyToHexIfEd25519OrNull(@Nullable final byte[] protobufKey)
+			throws InvalidProtocolBufferException {
+		if ((null == protobufKey) || (0 == protobufKey.length)) return null;
+
+		final var parsedKey = Key.parseFrom(protobufKey);
+		if (ED25519 != parsedKey.getKeyCase()) return null;
+
+		return Hex.encodeHexString(parsedKey.getEd25519().toByteArray());
+	}
```
Comment (gregscullard) [inline src/main/java/com/hedera/utilities/Utility.java]: add `.toLower()` just in case the protobuf key value is not lowercase

File: src/main/resources/db/migration/V1.11.0__entity_key.sql [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+-- NOTE: if the DB has many entities, this may need to be broken up into batches.
+update t_entities
+    set ed25519_public_key_hex = substring(encode(key::bytea, 'hex'), 4) -- This is where the key data starts in the protobuf
+    where key is not null
+        and encode(key::bytea, 'hex') like '1220%'; -- This indicates it's a single ED25519 key of proper length.
+
+--
+-- Update affected functions.
+--
+drop function if exists f_entity_create;
```
Comment (gregscullard) [inline src/main/resources/db/migration/V1.11.0__entity_key.sql]: Drop function requires the parameters of the function on some version of postgres if memory serves me right.

```
DROP FUNCTION if exists f_entity_create (
     _shard t_entities.entity_shard%TYPE
     ,_realm t_entities.entity_realm%TYPE
     ,_num t_entities.entity_num%TYPE
     ,_type_id t_entities.fk_entity_type_id%TYPE
     ,_exp_time_sec t_entities.exp_time_seconds%TYPE
     ,_exp_time_nanos t_entities.exp_time_nanos%TYPE
     ,_exp_time_ns t_entities.exp_time_ns%TYPE
     ,_auto_renew t_entities.auto_renew_period%TYPE
     ,_ed25519_public_key_hex t_entities.ed25519_public_key_hex%type
     ,_key t_entities.key%type
     ,_proxy_acc_id t_entities.fk_prox_acc_id%TYPE
 )
```

File: src/main/java/com/hedera/utilities/Utility.java (hunk 1 above)
Comment (mike-burrage-hedera) [reply src/main/java/com/hedera/utilities/Utility.java]: Done

File: src/main/java/com/hedera/recordFileLogger/Entities.java [hunk 3]
```diff
@@ -378,7 +396,7 @@ private long createOrGetEntity(long shard, long realm, long num, int fk_entity_t
 		entityCreate.setLong(7, 0);
     	entityCreate.setLong(8, 0);
 		entityCreate.setLong(9, 0);
-		entityCreate.setBytes(10, new byte[0]);
+		entityCreate.setNull(10, VARCHAR);
 		entityCreate.setBytes(11, new byte[0]);
```
[the remark this answers is not in the document]
Comment (mike-burrage-hedera) [reply src/main/java/com/hedera/recordFileLogger/Entities.java]: Done

File: src/main/java/com/hedera/recordFileLogger/Entities.java [hunk 4]
```diff
@@ -329,15 +340,22 @@ private long createEntity(long shard, long realm, long num, long exp_time_second
...
 
-		if (admin_key == null) {
-    		entityCreate.setBytes(10, new byte[0]);
-    	} else {
-    		entityCreate.setBytes(10, admin_key);
-    	}
-
     	if (key == null) {
+    		entityCreate.setNull(10, VARCHAR);
     		entityCreate.setBytes(11, new byte[0]);
```
[the remark this answers is not in the document]
Comment (mike-burrage-hedera) [reply src/main/java/com/hedera/recordFileLogger/Entities.java]: Done

File: src/main/resources/db/migration/V1.11.0__entity_key.sql (hunk 2 above)
Comment (mike-burrage-hedera) [reply src/main/resources/db/migration/V1.11.0__entity_key.sql]: Done.

**Comment by mike-burrage-hedera** [comment]
Can't have a unit test where utility.protobufKeyToHexIfEd25519OrNull() produces uppercase, since it always produces lowercase, but I updated the javadoc on it to make that explicit.

File: src/test/java/com/hedera/utilities/UtilityTest.java [hunk 5]
```diff
@@ -123,4 +130,39 @@ void getResource() {
 		assertThat(Utility.getResource("log4j2.xml")).exists().canRead();
 		assertThat(Utility.getResource("log4j2-test.xml")).exists().canRead();
 	}
+
+	private Utility getCut() throws SQLException { return new Utility(); }
+
+	@Test
+	@DisplayName("null")
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/utilities/UtilityTest.java]: DisplayName should include more context about what method you're testing as this will just show "null" in IDEs.

File: src/main/java/com/hedera/utilities/Utility.java [hunk 6]
```diff
@@ -745,4 +749,20 @@ public static File getResource(String path) {
...
 		}
 	}
+
+	/**
+	 * If the protobuf encoding of a Key is a single ED25519 key, return the key as a String with lowercase hex encoding.
+	 * @param protobufKey
+	 * @return ED25519 public key as a String in hex encoding, or null
+	 * @throws InvalidProtocolBufferException if the protobufKey is not a valid protobuf encoding of a Key (BasicTypes.proto)
+	 */
+	public @Nullable String protobufKeyToHexIfEd25519OrNull(@Nullable final byte[] protobufKey)
```
[the remark this answers is not in the document]
Comment (mike-burrage-hedera) [reply src/main/java/com/hedera/utilities/Utility.java]: Done

File: src/test/java/com/hedera/utilities/UtilityTest.java (hunk 5 above)
Comment (mike-burrage-hedera) [reply src/test/java/com/hedera/utilities/UtilityTest.java]: Done.

**Comment by gregscullard** [comment]
> Can't have a unit test where utility.protobufKeyToHexIfEd25519OrNull() produces uppercase, since it always produces lowercase, but I updated the javadoc on it to make that explicit.

I meant provide it an uppercase input, validate output is lowercase.

**Comment by mike-burrage-hedera** [comment]
There is no uppercase or lowercase input - the input is raw binary bits of the protobuf that
